server/services/prisonerService.js

```diff
@@ -149,6 +161,16 @@ function createPrisonerService(nomisClientBuilder) {
     }
   }
 
+  async function selectRo(allRoDetails) {
+    const roWithDeliusIds = allRoDetails.filter(r => !isEmpty(r.deliusId))
+
+    if (isEmpty(roWithDeliusIds)) {
+      throw new RoRelationshipError('No RO with a Delius staff code')
+    }
+
+    return roWithDeliusIds.sort((a, b) => b.personId - a.personId)[0]
+  }
+
```

Comment:
discard results without a delius id, and from what's left sort by person id to get the most recent person